src/@types/InviteCodes.ts

```diff
@@ -0,0 +1,10 @@
+import { Timestamp } from './Timestamp'
+
+/** An object that contains an invite code. */
+export interface InviteCodes {
+  id?: string // db only
+  created?: Timestamp
+  createdBy?: string
+  used?: Timestamp
+  usedBy?: string
+}
```

Comment:
I am not able to make this required as it will fail when user code is not there and we need to create. Currently I am creating codes, when user go to invite code modal. Maybe I should generate the code as soon as they register. 🤔